tests/acquisition/covidcast/test_csv_importer.py

```diff
@@ -0,0 +1,218 @@
+"""Unit tests for csv_importer.py."""
+
+# standard library
+import unittest
+from unittest.mock import MagicMock
+
+# third party
+import pandas
+
+# py3tester coverage target
+__test_target__ = 'delphi.epidata.acquisition.covidcast.csv_importer'
+
+
+class UnitTests(unittest.TestCase):
+  """Basic unit tests."""
+
+  def test_is_sane_day(self):
+    """Sanity check some dates."""
+
+    self.assertTrue(CsvImporter.is_sane_day(20200418))
+
+    self.assertFalse(CsvImporter.is_sane_day(22222222))
+    self.assertFalse(CsvImporter.is_sane_day(20200001))
+    self.assertFalse(CsvImporter.is_sane_day(20200199))
+    self.assertFalse(CsvImporter.is_sane_day(202015))
+
+  def test_is_sane_week(self):
+    """Sanity check some weeks."""
+
+    self.assertTrue(CsvImporter.is_sane_week(202015))
+
+    self.assertFalse(CsvImporter.is_sane_week(222222))
+    self.assertFalse(CsvImporter.is_sane_week(202000))
+    self.assertFalse(CsvImporter.is_sane_week(202054))
+    self.assertFalse(CsvImporter.is_sane_week(20200418))
+
+  def test_find_csv_files(self):
+    """Recursively explore and find CSV files."""
+
+    path_prefix = 'prefix/to/the/data/'
+    glob_paths = [
+      # valid weekly
+      path_prefix + 'fb_survey/weekly_202015_county_cli.csv',
+      # valid daily
+      path_prefix + 'ght/20200408_state_rawsearch.csv',
+      # invalid
+      path_prefix + 'invalid/hello_world.csv',
+      # invalid day
+      path_prefix + 'invalid/22222222_b_c.csv',
+      # invalid week
+      path_prefix + 'invalid/weekly_222222_b_c.csv',
+      # invalid geography
+      path_prefix + 'invalid/20200418_country_c.csv',
+      # ignored
+      path_prefix + 'ignored/README.md',
+    ]
+    mock_glob = MagicMock()
+    mock_glob.glob.return_value = glob_paths
+
+    found = list(CsvImporter.find_csv_files(path_prefix, glob=mock_glob))
+
+    expected = [
+      (glob_paths[0], ('fb_survey', 'cli', 'week', 'county', 202015)),
+      (glob_paths[1], ('ght', 'rawsearch', 'day', 'state', 20200408)),
+      (glob_paths[2], None),
+      (glob_paths[3], None),
+      (glob_paths[4], None),
+      (glob_paths[5], None),
+    ]
+    self.assertEqual(found, expected)
+
+  def test_is_header_valid_allows_extra_columns(self):
```

Comment:
maybe also a test that column order doesn't matter? but maybe it's not necessary if the risk of someone switching to a non-set-based implementation later is low.

Reply:
done